File: backend/app/auth/schemas.py
"""
Pydantic schemas for authentication endpoints.
"""
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, EmailStr, Field


class UserSignUp(BaseModel):
    """Schema for user registration."""
    email: EmailStr
    password: str = Field(..., min_length=8, description="Password must be at least 8 characters")
    account_type: Literal["job_seeker", "employer"] = Field(..., description="Account type")
    full_name: str | None = Field(None, description="User's full name")


class UserSignIn(BaseModel):
    """Schema for user login."""
    email: EmailStr
    password: str


class TokenResponse(BaseModel):
    """Schema for authentication token response."""
    access_token: str
    token_type: str = "bearer"
    expires_in: int
    refresh_token: str
    user: "UserInfo"


class UserInfo(BaseModel):
    """Schema for user information."""
    id: str
    email: str
    account_type: str | None = None
    email_verified: bool = False
    provider: str = "email"
    created_at: datetime | None = None


class PasswordReset(BaseModel):
    """Schema for password reset request."""
    email: EmailStr


class PasswordUpdate(BaseModel):
    """Schema for password update."""
    password: str = Field(..., min_length=8, description="New password must be at least 8 characters")


class CurrentUser(BaseModel):
    """Schema for current authenticated user."""
    id: str
    email: str
    account_type: str | None = None
    provider: str
    email_verified: bool
    role: str
    metadata: dict = {}


class MessageResponse(BaseModel):
    """Schema for simple message responses."""
    message: str

File: frontend/src/lib/api/auth/schemas.ts
/**
 * Zod schemas for authentication API
 * These match the backend Pydantic schemas in backend/app/auth/auth_schemas.py
 */
import { z } from "zod";

// UserSignUp schema - matches backend UserSignUp
export const UserSignUpSchema = z.object({
  email: z.string().email("Invalid email address"),
  password: z.string().min(8, "Password must be at least 8 characters"),
  account_type: z.enum(["job_seeker", "employer"], {
    errorMap: () => ({
      message: "Account type must be 'job_seeker' or 'employer'",
    }),
  }),
  full_name: z.string().nullable().optional(),
});

// UserSignIn schema - matches backend UserSignIn
export const UserSignInSchema = z.object({
  email: z.string().email("Invalid email address"),
  password: z.string().min(1, "Password is required"),
});

// UserInfo schema - matches backend UserInfo
export const UserInfoSchema = z.object({
  id: z.string(),
  email: z.string(),
  account_type: z.string().nullable().optional(),
  email_verified: z.boolean().optional().default(false),
  provider: z.string().optional().default("email"),
  created_at: z.string().nullable().optional(),
});

// TokenResponse schema - matches backend TokenResponse
export const TokenResponseSchema = z.object({
  access_token: z.string(),
  token_type: z.string().default("bearer"),
  expires_in: z.number(),
  refresh_token: z.string(),
  user: UserInfoSchema,
});

// CurrentUser schema - matches backend CurrentUser
export const CurrentUserSchema = z.object({
  id: z.string(),
  email: z.string(),
  account_type: z.string().nullable().optional(),
  provider: z.string(),
  email_verified: z.boolean(),
  role: z.string(),
  metadata: z.record(z.unknown()).optional().default({}),
});

// Email confirmation response schema (alternative response from register endpoint)
export const EmailConfirmationResponseSchema = z.object({
  message: z.string(),
  user_id: z.string(),
  email: z.string(),
  email_confirmation_required: z.literal(true),
});

// MessageResponse schema - matches backend MessageResponse
export const MessageResponseSchema = z.object({
  message: z.string(),
});
